lib/cli-config.js

```diff
@@ -70,12 +72,16 @@ exports.getContent = function(config, directory) {
 
         if (ext === '.js' || ext === '.json') {
             content = require(configPath);
+        } else if (ext === '.yaml') {
+            content = yaml.safeLoad(fs.readFileSync(configPath, 'utf8'));
         } else {
-            content = JSON.parse(
-                stripJSONComments(
-                    fs.readFileSync(configPath, 'utf8')
-                )
-            );
+            data = fs.readFileSync(configPath, 'utf8');
+
+            try {
+                content = JSON.parse(stripJSONComments(data));
+            } catch (e) {
+                content = yaml.safeLoad(data);
```

Comment:
I wouldn't say it is out of scope, different commit maybe but not a pr, since this pr would change error output, from json error to custom one, with jsonlint we could preserve and improve that message rather then regress it